### PR TITLE
add NullTransport

### PR DESCRIPTION
**Pull request by garak**
See issue #9

### CODE REVIEW

File: lib/Transport/NullTransport.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+/**
+ * Null Transport.
+ * Useful for testing purposes.
+ *
+ * @author Massimiliano Arione <garakkio@gmail.com>
+ */
+class NullTransport implements TransportInterface
+{
+    public function executeRequest(string $uri, string $body): string
```
Comment (massimilianobraglia) [inline lib/Transport/NullTransport.php]: Can you add the phpdoc here? 
```php
/**
  * {@inheritdoc}
  */
public function executeRequest(string $uri, string $body): string
```

**Review by massimilianobraglia — COMMENTED** [review]
Thank you, @garak!
I have left just a little comment.

File: lib/Transport/NullTransport.php (hunk 1 above)
Comment (garak) [reply lib/Transport/NullTransport.php]: IMHO such phpdoc is useless once you got full type-hinting. Anyway, I understand the request as a matter on consistence with other parts of code (but, again: you should instead remove it everywhere)

**Comment by alekitto** [comment]
IMHO TestTransport is a better name than NullTransport. Could be used to fake the response and check an application behavior.

**Comment by garak** [comment]
@alekitto just because you can use it outside tests (e.g. in dev environment) I named it "null".
Also, this is consistent with similar uses in other packages (e.g. Swifmailer)

File: lib/Transport/NullTransport.php (hunk 1 above)
Comment (massimilianobraglia) [reply lib/Transport/NullTransport.php]: It is not only for typehintings: if the docs for a method does not need some more explanation from the one defined in the interface we inherit that docs. This is the way we use it 😄

**Review by massimilianobraglia — APPROVED** [review]

**Comment by massimilianobraglia** [comment]
I agree with @garak in this context. Another example is the `NullTransport` of Elastica (https://github.com/ruflin/Elastica/blob/master/lib/Elastica/Transport/NullTransport.php)

**Comment by alekitto** [comment]
Null implies a void response to be generated. Here there is no void response, but a fake response, with fake data.
It implies you used test classic report method, it fakes the message id and the remaining messages count.
It is not null or empty, it’s just fake.

**Comment by garak** [comment]
@alekitto you're right. I started trying with an  empty response, but unfortunately it didn't work: the method **must** return a string, and such string is expected to be something understandable.
Maybe we could improve the fake response to match some input parameters?

**Comment by alekitto** [comment]
@garak yes, maybe could be setted (or injected in construction) overriding some defaults. This way no parameter must be set if no check are done, but can be overridden if needed

**Comment by garak** [comment]
@alekitto I guess that your suggestion could be a matter for an eventual improvement. For my (very limited) requirements, current dummy implementation is fine

**Comment by alekitto** [comment]
That’s why i suggested a change of the class name. DummyTransport could be an alternative, considering the impossibility to represent an “empty” response

**Comment by massimilianobraglia** [comment]
I like the `DummyTransport` name. It is a good compromise. What do you think about it, @garak?

**Review by massimilianobraglia — APPROVED** [review]

**Comment by massimilianobraglia** [comment]
Thank you, @garak!

**Comment by garak** [comment]
@massimilianobraglia what about moving alias to `0.3` and releasing `0.2`?

**Comment by massimilianobraglia** [comment]
https://github.com/fazland/skebby-rest-client/releases/tag/0.2.0
here you are!

**Pull request merged**